homework/src/index.js

```diff
@@ -31,17 +33,124 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+    const select = createAndAppend('select', header, { id: 'selectMenu' });
+    // use promise object
+    const promise = fetchJSON(url);
+    promise.then(data => {
+      data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+      data.forEach((repository, index) => {
+        createAndAppend('option', select, {html: repository.name, value: index});
+      });
+      
+      const container = createAndAppend('div', root, { id: 'container'});
+      contentLeft(container, data, select);
+      contentRight(container, data, select);
+    })
+    .catch(error => {
+      createAndAppend('div', root, { html: error.message, class: 'alert-error' });
+    });
+  }
+
+
+
+
+
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
```

Comment:
I would rename `data` to `repositories`. In general, use generic, meaningless names such as `data`, `info` etc only when the variable or parameter can contain anything. When you know, as in this case, exactly what the function parameter will receive, choose a names that accurately describes it contents.